### PR TITLE
Nav Menu Upgrades and Minor UX Improvements

### PR DESCRIPTION
**Pull request by MythicalCow**
### Change List 🔨 
* Converted navigation menu into a dropdown menu
* Moved collapsed side panel menu icons into top row
* Auto refresh when conversation is deleted to update side panel and route back to main page if deletion is on current conversation
* Highlight the current conversation in the side panel
* Dynamic homepage messages with current day and time of day.
* `colorutils` upgraded to have more expansive tailwind color options and dynamic class name generation.
* Converted create agent button alert into shadcn `ToolTip`
* Colored lines and icons for agents in chat window
* Cleaned up border styling in dark mode
* fixed three dot menu in side panel to be more easier to click

### Image 🖼️ 
image

### Feedback 🤝 
* Design comments
* Flow between pages
* Bugs and edge case testing

### CODE REVIEW

File: src/interface/web/app/agents/page.tsx [hunk 1]
```diff
@@ -318,41 +266,56 @@ export default function Agents() {
...
+            <div className={`${styles.pageLayout} w-full ml-auto mr-auto`}>
+                <div className={`${styles.sidePanel} top-0`}>
+                    <SidePanel
+                        webSocketConnected={true}
+                        conversationId={null}
+                        uploadedFiles={[]}
+                        isMobileWidth={isMobileWidth}
+                    />
+                </div>
+                <div className={`ml-auto mr-auto ${isMobileWidth ? "w-11/12" : "w-1/2"} pt-4`}>
```
Comment (debanjum) [inline src/interface/web/app/agents/page.tsx:288]: Can ml-auto and mr-auto be replace with just `mx-auto` (which should take care of x-axis margin)?
```suggestion
                <div className={`mx-auto ${isMobileWidth ? "w-11/12" : "w-1/2"} pt-4`}>
```

File: src/interface/web/app/agents/page.tsx [hunk 2]
```diff
@@ -318,41 +266,56 @@ export default function Agents() {
...
+                    <div className="pt-8 flex">
+                        <h1 className="text-3xl">Agents</h1>
+                        <div className="ml-auto float-right border pt-3 pb-2 pl-2 pr-2 rounded-xl font-bold hover:bg-stone-100 dark:hover:bg-neutral-900">
```
Comment (debanjum) [inline src/interface/web/app/agents/page.tsx:291]:
```suggestion
                        <div className="ml-auto float-right border p-2 pt-3 rounded-xl font-bold hover:bg-stone-100 dark:hover:bg-neutral-900">
```

File: src/interface/web/app/common/colorUtils.ts [hunk 3]
```diff
@@ -1,40 +1,16 @@
...
-    if (color === 'indigo') return `bg-gradient-to-b from-[hsl(var(--background))] to-indigo-100/70 dark:from-[hsl(var(--background))] dark:to-indigo-950/30 `;
-    if (color === 'fuschia') return `bg-gradient-to-b from-[hsl(var(--background))] to-fuschia-100/70 dark:from-[hsl(var(--background))] dark:to-fuschia-950/30 `;
-    if (color === 'rose') return `bg-gradient-to-b from-[hsl(var(--background))] to-rose-100/70 dark:from-[hsl(var(--background))] dark:to-rose-950/30 `;
-    if (color === 'sky') return `bg-gradient-to-b from-[hsl(var(--background))] to-sky-100/70 dark:from-[hsl(var(--background))] dark:to-sky-950/30 `;
-    if (color === 'amber') return `bg-gradient-to-b from-[hsl(var(--background))] to-amber-100/70 dark:from-[hsl(var(--background))] dark:to-amber-950/30 `;
-    if (color === 'emerald') return `bg-gradient-to-b from-[hsl(var(--background))] to-emerald-100/70 dark:from-[hsl(var(--background))] dark:to-emerald-950/30 `;
+    if (tailwindColors.includes(color)) {
+        return `bg-gradient-to-b from-[hsl(var(--background))] to-${color}-100/70 dark:from-[hsl(var(--background))] dark:to-${color}-950/30 `;
```
Comment (debanjum) [inline src/interface/web/app/common/colorUtils.ts:12]: Nice! Reduce code duplication

File: src/interface/web/app/agents/page.tsx [hunk 4]
```diff
@@ -318,41 +266,56 @@ export default function Agents() {
...
-                    }
-                    <div className='mt-3' />
-                    <strong>Coming Soon:</strong> Support for making your own agents.
-                </AlertDescription>
-            </Alert>
-            <div className={styles.agentList}>
-                {data.map(agent => (
-                    <AgentCard key={agent.slug} data={agent} userProfile={authenticatedData} isMobileWidth={isMobileWidth} />
-                ))}
+            <div className={`${styles.pageLayout} w-full ml-auto mr-auto`}>
```
Comment (debanjum) [inline src/interface/web/app/agents/page.tsx:279]:
```suggestion
            <div className={`${styles.pageLayout} w-full mx-auto`}>
```

File: src/interface/web/app/agents/page.tsx [hunk 5]
```diff
@@ -318,41 +266,56 @@ export default function Agents() {
     }
 
     return (
-        <main className={styles.main}>
-            <h3
-                className='text-xl py-4'>
-                Agents
-            </h3>
+        <main className={`${styles.main} w-full ml-auto mr-auto`}>
```
Comment (debanjum) [inline src/interface/web/app/agents/page.tsx:269]:
```suggestion
        <main className={`${styles.main} w-full mx-auto`}>
```

**Review by debanjum — COMMENTED** [review]
Haven't tested or done a proper review. Left some minor comments but otherwise looks good to me

**Review by sabaimran — APPROVED** [review]
Awesome! with this set of changes it's all coming together. 🫶🏾

**Pull request merged**